sdk/resourcemanager/azure-resourcemanager-cdn/src/main/java/com/azure/resourcemanager/cdn/implementation/CdnStandardRulesEngineRuleImpl.java

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.resourcemanager.cdn.implementation;
+
+import com.azure.resourcemanager.cdn.models.CdnStandardRulesEngineRule;
+import com.azure.resourcemanager.cdn.models.CdnEndpoint;
+import com.azure.resourcemanager.cdn.models.DeliveryRule;
+import com.azure.resourcemanager.cdn.models.DeliveryRuleAction;
+import com.azure.resourcemanager.cdn.models.DeliveryRuleCondition;
+import com.azure.resourcemanager.resources.fluentcore.arm.models.implementation.ChildResourceImpl;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Implementation for {@link CdnStandardRulesEngineRule}.
+ */
+class CdnStandardRulesEngineRuleImpl
+    extends ChildResourceImpl<DeliveryRule, CdnEndpointImpl, CdnEndpoint>
+    implements CdnStandardRulesEngineRule,
+    CdnStandardRulesEngineRule.Definition<CdnEndpointImpl>,
+    CdnStandardRulesEngineRule.Update<CdnEndpointImpl> {
+
+    CdnStandardRulesEngineRuleImpl(CdnEndpointImpl parent, String name) {
+        this(parent, new DeliveryRule().withName(name));
+    }
+
+    CdnStandardRulesEngineRuleImpl(CdnEndpointImpl parent, DeliveryRule deliveryRule) {
+        super(deliveryRule, parent);
+    }
+
+    @Override
+    public CdnStandardRulesEngineRuleImpl withOrder(int order) {
+        innerModel().withOrder(order);
+        return this;
+    }
+
+    @Override
+    public CdnStandardRulesEngineRuleImpl withMatchConditions(DeliveryRuleCondition... matchConditions) {
+        List<DeliveryRuleCondition> conditions = new ArrayList<>();
+        if (matchConditions != null) {
+            conditions.addAll(Arrays.asList(matchConditions));
+        }
+        innerModel().withConditions(conditions);
+        return this;
+    }
+
+    @Override
+    public CdnStandardRulesEngineRuleImpl withActions(DeliveryRuleAction... actions) {
+        List<DeliveryRuleAction> actionList = new ArrayList<>();
+        if (actions != null) {
+            actionList.addAll(Arrays.asList(actions));
+        }
+        innerModel().withActions(actionList);
+        return this;
+    }
```

Comment:
I'm not sure, will check it. Thanks!

Reply:
From [docs](https://learn.microsoft.com/en-us/azure/cdn/cdn-standard-rules-engine-reference#terminology) and Portal:

`actions` is always required.

For `matchConditions`:
1. for Global rule(order=0), it must be left blank, and makes sense since it's global and always will be triggered
2. for other rules(order=1,2,3,...,etc), it's required

two alternatives for Global rule:
1. introduce `defineNewStandardRulesEngineGlobalRule().withActions(actions)` and `withoutStandardRulesEngineGlobalRule()`.
2. introduce `withoutMatchConditions().withActions(actions)`.

Personally I prefer the first alternative.

Reply:
Maybe make the MatchCondition stage optional? Then, user can just do `withActions(...).attach()`.

Reply:
Sure, sounds good!